rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,521 @@
+        self.eth = A_ETH.resolve_to_crypto_asset()
+        self.sai = A_SAI.resolve_to_evm_token()
+        self.peth = Asset(f'eip155:1/erc20:{POOLED_ETHER_ADDRESS}')
+
+    def _decode_makerdao_sai_tub(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],
+            action_items: Optional[list[ActionItem]],
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        if tx_log.topics[0] == MAKERDAO_SAITUB_NEW_CDP_TOPIC:
+            return self._decode_makerdao_new_cdp_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_CLOSE_CDP_TOPIC:
+            return self._decode_makerdao_close_cdp(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_BORROW_SAI_TOPIC:
+            return self._decode_makerdao_borrow_sai_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_REPAY_SAI_TOPIC:
+            return self._decode_makerdao_repay_sai_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_DEPOSIT_WETH_TOPIC:
+            return self._decode_makerdao_deposit_weth_eth_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_LIQUIDATE_TOPIC:
+            return self._decode_makerdao_liquidate_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_WITHDRAW_TOPIC:
+            return self._decode_makerdao_withdraw_eth(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+
+        return None, []
+
+    def _decode_makerdao_withdraw_eth(
```

Comment:
Here and in the other functions. No need to repeat "makerdao". This is a makerdao class.

```suggestion
    def _decode_withdraw_eth(
```